src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8879,6 +8892,11 @@ private ArgumentSyntax ParseArgumentExpression(bool isIndexer)
                 expression = this.ParseSubExpression(0);
             }
 
+            if (refOrOutKeyword != null && IsLambda(expression.Kind))
+            {
+                expression = this.AddError(expression, ErrorCode.ERR_RefLvalueExpected);
```

Comment:
This is actually already dealt with, but only enabled using the strict feature. Will remove fix, but update tests for now.